frontend/src/metabase/static-viz/components/LineAreaBarChart/utils/series.tsx

```diff
@@ -0,0 +1,174 @@
+import _ from "underscore";
+import { merge } from "icepick";
+import { isNotNull } from "metabase/core/utils/types";
+import { getColorsForValues } from "metabase/lib/colors/charts";
+import { formatStaticValue } from "metabase/static-viz/lib/format";
+import { ColorPalette } from "metabase/lib/colors/types";
+import {
+  ChartSettings,
+  Series,
+  SeriesWithOneOrLessDimensions,
+  SeriesWithTwoDimensions,
+} from "../../XYChart/types";
+
+export function getSeriesWithColors(
+  multipleSeries: (SeriesWithOneOrLessDimensions | SeriesWithTwoDimensions)[][],
+  settings: ChartSettings,
+  palette: ColorPalette,
+): (SeriesWithOneOrLessDimensions | SeriesWithTwoDimensions)[][] {
+  const isMultipleSeries = multipleSeries.length > 1;
+  const keys = getSeriesKeys(multipleSeries, isMultipleSeries);
+
+  const seriesColors = settings.series_settings
+    ? _.mapObject(settings.series_settings, value => {
+        return value.color;
+      })
+    : undefined;
```

Comment:
This achieve the same thing as 
https://github.com/metabase/metabase/blob/cc589cf0f6a3f4dfe8e22aee2c1f21a8a595ddd0/frontend/src/metabase/visualizations/lib/settings/series.js#L178-L182

but with simpler logic.